lib/elixir/lib/access.ex

```diff
@@ -135,6 +152,14 @@ defmodule Access do
         # Implementation of the Access callbacks...
       end
 
+  **Note:** When implementing the behaviour, Access requires any key comparisons to be
+  implemented using the `===` operator.
+
+  ### Why not a Protocol?
+
+  Previously Access was expressed as a Protocol, but as it is used in scripts and in the
```

Comment:
Our API documentations typically does not include implementation decisions. For example, why HashDict used tuples of tuples as trees with a factor of 16? So I am a bit torn on having this discussion here.